OneCommunityGlobal/HighestGoodNetworkApp
# Ruike enforce no space in file name

# Description
Remove all files with filename contains a white space, and enforce rule by setting .gitignore

## Related PRS (if any):
N/A

## Main changes explained:
Changed several files' filename, replace ' ' with '_'
add **\ ** in gitignore to ignore changes with whitespace in the filename

## How to test:
1. check into current branch
2. do `npm install` (optional)
3. go to the parent directory of frontend/backend folder, for example, the Github folder
4. run the following python file and make sure no file path was printed
```
import os

backend = "HGNRest"
frontend = "HighestGoodNetworkApp"

def checkfile(folderName):
        for fileName in os.scandir(folderName):
                if os.path.isfile(fileName) and ' ' in os.path.basename(fileName):
                        print(os.path.relpath(fileName))
                elif os.path.isdir(fileName):
                        checkfile(fileName)

checkfile(backend)
checkfile(frontend)

```
5. can also test by creating new file with filename contains ' ', check that there's no change in git


## Screenshots or videos of changes:
add a new file that contains whitespace in filename
<img width="255" alt="Screen Shot 2023-10-25 at 8 41 36 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/119435da-09ad-496b-bc29-aa97a06e13e3">
i was using github desktop, make sure there's no local change
<img width="523" alt="Screen Shot 2023-10-25 at 8 42 48 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/ee4a70e3-a1bc-48ed-95ae-f561e21e78d5">


## Note:
Make sure when run the script, the frontend/backend folder are in the same directory
such as 
|| /Github
||  => *.py(replace * with any name)
||  => /frontend
||  => /backend
and run python3 *.py(replace * with any name)
